test/test_grpc.py

```diff
@@ -46,86 +46,50 @@ class TestCreateStub(unittest2.TestCase):
     FAKE_SERVICE_PATH = 'service_path'
     FAKE_PORT = 10101
 
-    @mock.patch('grpc.composite_channel_credentials')
-    @mock.patch('grpc.ssl_channel_credentials')
-    @mock.patch('grpc.secure_channel')
-    @mock.patch('google.gax.auth.make_auth_func')
-    def test_creates_a_stub_ok_with_no_scopes(
-            self, auth, chan, chan_creds, comp):
+    @mock.patch('google.gax.auth.get_default_credentials')
+    @mock.patch('google.gax.auth.secure_authorized_channel')
+    def test_creates_a_stub_with_default_credentials(
+            self, secure_authorized_channel, get_default_credentials):
+        fake_scopes = ['one', 'two']
         got_channel = grpc.create_stub(
-            _fake_create_stub, self.FAKE_SERVICE_PATH, self.FAKE_PORT)
-        chan_creds.assert_called_once_with()
-        chan.assert_called_once_with(
+            _fake_create_stub, service_path=self.FAKE_SERVICE_PATH,
+            service_port=self.FAKE_PORT, scopes=fake_scopes)
+
+        get_default_credentials.assert_called_once_with(fake_scopes)
+        secure_authorized_channel.assert_called_once_with(
+            get_default_credentials.return_value,
             '{}:{}'.format(self.FAKE_SERVICE_PATH, self.FAKE_PORT),
-            comp.return_value)
-        auth.assert_called_once_with([])
-        self.assertEqual(got_channel, chan.return_value)
-
-    @mock.patch('grpc.composite_channel_credentials')
-    @mock.patch('grpc.ssl_channel_credentials')
-    @mock.patch('grpc.secure_channel')
-    @mock.patch('google.gax.auth.make_auth_func')
-    def test_creates_a_stub_ok_with_scopes(
-            self, auth, chan, chan_creds, comp):
-        fake_scopes = ['dummy', 'scopes']
-        grpc.create_stub(
-            _fake_create_stub, self.FAKE_SERVICE_PATH, self.FAKE_PORT,
-            scopes=fake_scopes)
-        chan_creds.assert_called_once_with()
-        chan.assert_called_once_with(
+            ssl_credentials=None)
+
+        self.assertEqual(got_channel, secure_authorized_channel.return_value)
+
+    @mock.patch('google.gax.auth.get_default_credentials')
+    @mock.patch('google.gax.auth.secure_authorized_channel')
+    def test_creates_a_stub_with_explicit_credentials(
+            self, secure_authorized_channel, get_default_credentials):
+        credentials = mock.Mock()
+        got_channel = grpc.create_stub(
+            _fake_create_stub, service_path=self.FAKE_SERVICE_PATH,
+            service_port=self.FAKE_PORT, credentials=credentials)
+
+        self.assertFalse(get_default_credentials.called)
```

Comment:
`create_stub` has a branch that calls `get_default_credentials`, this assert ensures that branch was not executed.